PrepatcherCodeGenerator/PlayerDataAccessGenerator.cs

```diff
@@ -0,0 +1,153 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+
+namespace PrepatcherCodeGenerator;
+
+[Generator]
+public class PlayerDataAccessGenerator : IIncrementalGenerator
+{
+    [SuppressMessage("MicrosoftCodeAnalysisReleaseTracking", "RS2008:Enable analyzer release tracking", Justification = "No release")]
+    private static readonly DiagnosticDescriptor noPlayerDataDescriptor = new(
+        id: "PDA001",
+        title: "Player Data Missing",
+        messageFormat: "Could not find the PlayerData symbol in the compilation. The generator cannot run.",
+        category: "SourceGeneration",
+        defaultSeverity: DiagnosticSeverity.Error,
+        isEnabledByDefault: true);
+
+    private bool IsValidField(IFieldSymbol symbol)
+    {
+        if (symbol.IsStatic || symbol.IsReadOnly)
+        {
+            return false;
+        }
+
+        if (symbol.DeclaredAccessibility != Accessibility.Public)
+        {
+            return false;
+        }
+        // We assume that the publicizer doesn't make things non-public, so we don't need to check the attribute param
```

Comment:
After further thought I think it doesn't cost much to not need to make this assumption, so probably a good idea to check it. I don't think it would be entirely unreasonable that they just apply this to everything, even if they currently don't

Reply:
True